compiler/rustc_mir_transform/src/lib.rs

```diff
@@ -548,6 +549,8 @@ fn run_optimization_passes<'tcx>(tcx: TyCtxt<'tcx>, body: &mut Body<'tcx>) {
             &uninhabited_enum_branching::UninhabitedEnumBranching,
             &o1(simplify::SimplifyCfg::AfterUninhabitedEnumBranching),
             &inline::Inline,
+            // Remove switches on a statically known discriminant, which can happen as a result of inlining.
+            &simplify_static_switch::SimplifyStaticSwitch,
```

Comment:
This pass clones statements. As we have SSA-based passes, and we don't try to recover SSAness, it should be run much later.

Reply:
i have moved the pass to right before `SeparateConstSwitch`.